jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2957,40 +2937,71 @@ code_blockt java_bytecode_convert_methodt::convert_astore(
 }
 
 optionalt<exprt> java_bytecode_convert_methodt::convert_invoke_dynamic(
-  const java_class_typet::java_lambda_method_handlest &lambda_method_handles,
   const source_locationt &location,
-  const exprt &arg0)
+  std::size_t instruction_address,
+  const exprt &arg0,
+  codet &result_code)
 {
   const java_method_typet &method_type = to_java_method_type(arg0.type());
+  const java_method_typet::parameterst &parameters(method_type.parameters());
+  const typet &return_type = method_type.return_type();
 
-  const optionalt<symbolt> &lambda_method_symbol = get_lambda_method_symbol(
-    lambda_method_handles,
-    method_type.get_int(ID_java_lambda_method_handle_index));
-  if(lambda_method_symbol.has_value())
-    debug() << "Converting invokedynamic for lambda: "
-            << lambda_method_symbol.value().name << eom;
-  else
-    debug() << "Converting invokedynamic for lambda with unknown handle "
-               "type"
-            << eom;
+  // Note these must be popped regardless of whether we understand the lambda
+  // method or not
+  code_function_callt::argumentst arguments = pop(parameters.size());
 
-  const java_method_typet::parameterst &parameters(method_type.parameters());
+  irep_idt synthetic_class_name =
+    lambda_synthetic_class_name(method_id, instruction_address);
 
-  pop(parameters.size());
+  if(!symbol_table.has_symbol(synthetic_class_name))
+  {
+    // We failed to parse the invokedynamic handle as a Java 8+ lambda;
```

Comment:
Should we output a warning in this case?

Reply:
A debug message is already produced in `java_bytecode_parser.cpp` in this case. That's probably enough for now, as otherwise we would print a warning for every non-Java use of invokedynamic, which would be pretty user-unfriendly